lib/error-cat.js

```diff
@@ -0,0 +1,99 @@
+'use strict'
+
+const RollbarReporter = require('./rollbar-reporter')
+
+/**
+ * Base class for the error-cat module. Provides methods that allow the end user
+ * to report errors, set default reporting levels, and change reporting engines.
+ * @class
+ * @author Ryan Sandor Richards
+ */
+class ErrorCat {
+  /**
+   * Creates a new ErrorCat instance.
+   * @param {error-cat:Reporter} [reporter] Reporter used by error-cat to report
+   *   errors. Uses the built-in rollbar reporter by default.
+   */
+  constructor (reporter) {
+    // Set the reporter delegate
+    if (!reporter) {
+      reporter = new RollbarReporter()
+      reporter.setReportingLevel('error')
+    }
+    this.setReporter(reporter)
+
+    // Add bound helper methods
+    this['catch'] = this._reportAndThrow.bind(this)
+    this['middleware'] = this._reportAndPass.bind(this)
+  }
+
+  /**
+   * Sets the reporter delegate. The delegate is used to actually perform the
+   * specifics of error reporting through error-cat.
+   * @see error-cat:reporters~AbstractReporter
+   * @param {AbstractReporter} reporter Reporter to use.
+   */
+  setReporter (reporter) {
```

Comment:
I'd be nicer if you could have multiple reporters (`console` & Rollbar) for example... but this is a future PR.